scripts/upgrade-clients.js

```diff
@@ -0,0 +1,100 @@
+const Role = require("./domain/Role");
+
+/**
+ * Upgrades clients
+ *
+ * Prerequisite:
+ * - Admin must have UPGRADER role. Use `manage-roles.js` to grant it.
+ *
+ * Currently script upgrades the only existing client - BosonVoucher.
+ * If new clients are introduced, this script should be modified to get the list of clients to upgrade from the config.
+ */
+async function main(env) {
+  // Bail now if hardhat network, unless the upgrade is tested
+  if (network === "hardhat" && env !== "upgrade-test") process.exit();
+
+  const { chainId } = await ethers.provider.getNetwork();
+  let { contracts } = readContracts(chainId, network, env);
+
+  const divider = "-".repeat(80);
+  console.log(`${divider}\nBoson Protocol Client Upgrader\n${divider}`);
+  console.log(`⛓  Network: ${network}\n📅 ${new Date()}`);
+
+  // If hardhat, get an address generated by the mnemonic
+  const adminAddress =
+    network === "hardhat" ? (await ethers.getSigners())[0].address : environments[network].adminAddress;
+
+  // If admin address is unspecified, exit the process
+  if (adminAddress == ethers.constants.AddressZero || !adminAddress) {
+    console.log("Admin address must not be zero address");
+    process.exit(1);
+  }
+
+  // Get list of accounts managed by node
+  const nodeAccountList = (await ethers.provider.listAccounts()).map((address) => address.toLowerCase());
+
+  if (nodeAccountList.includes(adminAddress.toLowerCase())) {
+    console.log("🔱 Admin account: ", adminAddress);
+  } else {
+    console.log("🔱 Admin account not found");
+    process.exit(1);
+  }
+  console.log(divider);
+
+  // Get addresses of currently deployed contracts
+  const accessControllerAddress = contracts.find((c) => c.name === "AccessController").address;
+  if (!accessControllerAddress) {
+    return addressNotFound("AccessController");
+  }
+  const beaconAddress = contracts.find((c) => c.name === "BosonVoucher Beacon").address;
+  if (!beaconAddress) {
+    return addressNotFound("BosonVoucher Beacon");
+  }
+
+  // Get AccessController abstraction
+  const accessController = await ethers.getContractAt("AccessController", accessControllerAddress);
+
+  // Check that caller has upgrader role.
+  const hasRole = await accessController.hasRole(Role.UPGRADER, adminAddress);
+  if (!hasRole) {
+    console.log("Admin address does not have UPGRADER role");
+    process.exit(1);
+  }
+
+  // Deploy Protocol Client implementation contracts
+  console.log(`\n📋 Deploying new logic contract`);
+  const [bosonVoucherImplementation] = await deployProtocolClientImpls(maxPriorityFeePerGas);
+
+  // Update implementation address on beacon contract
+  console.log(`\n📋 Updating implementation address on beacon`);
+  const beacon = await ethers.getContractAt("BosonClientBeacon", beaconAddress);
+  await beacon.setImplementation(bosonVoucherImplementation.address, await getFees(maxPriorityFeePerGas));
```

Comment:
This line fails when I run on localhost with an error saying the caller doesn't have the correct role. 

`  const adminSigner = await ethers.getSigner(adminAddress);`

should be called first, and then `beacon` should connect to `adminSigner` when calling `setImplementation`

Reply:
Fixed.